service/src/main/java/org/cbioportal/service/impl/StructuralVariantServiceImpl.java

```diff
@@ -113,17 +132,31 @@
         List<String> fusionVariantSampleIds = new ArrayList<String>();
         Map<String, String> molecularProfileIdReplaceMap = new HashMap<>();
 
+        // TODO: Remove once fusions are removed from mutation table
+        Map<String, MolecularProfile> molecularProfileMap = molecularProfileRepository
+                .getMolecularProfiles(new HashSet<>(molecularProfileIds), "DETAILED")
+                .stream()
+                .collect(Collectors.toMap(datum -> datum.getStableId(), datum -> datum));
+
         // TODO: Remove once fusions are removed from mutation table
         for (int i = 0; i < molecularProfileIds.size(); i++) {
             String molecularProfileId = molecularProfileIds.get(i);
-            if (molecularProfileId.endsWith("_structural_variants")) {
+            if (molecularProfileId.endsWith(molecularProfileUtil.STRUCTURAL_VARIANT_PROFILE_SUFFIX)) {
```

Comment:
👍  ... and if we are going to be recognizing the profile "kind" through stable id examination, maybe we should be encapsulating/collecting that logic into a utility class (MolecularProfileUtil?) and making a clearly named function call instead. Such as
molecularProfileUtil.isOfKindStructuralVariant(molecularProfileId)

Relying on the stable id to tell the profile kind/category seems like a bad idea in general. Once we get rid of the blending of fusions into mutations, I suggest we make an effort to stop doing this -- or that we formalize rules for the naming of profiles and enforce those rules during import (i.e. prohibit importing a mutation profile with a stable id which does not match "_mutations").

Reply:
hopefully we'll be able to just drop that completely when fusions are removed from the mutations table